### ISSUE TITLE
Buf code generation using remote plugins

### ISSUE THREAD

**Issue by nickjn92** [andrewparmet/buf-gradle-plugin#37]
I've understand that a task like `bufGenerate` is not currently available because it relies on having a protoc available.
However, now buf supports using remote plugins to generate code, hence no need for protoc to be available locally.

With the presence of a `buf.gen.yaml` file (note managed part is optional) you can generate code using remote plugins instead:

```yaml
version: v1

managed:
  enabled: true
  java_multiple_files: true
  java_package_prefix: se

plugins:
  - remote: buf.build/protocolbuffers/plugins/kotlin:v3.19.3-1
    out: build/gen/kotlin
  - remote: buf.build/grpc/plugins/kotlin:v1.2.1-1
    out: build/gen/kotlin
```

This would be a fantastic addition to the plugin as we could use this instead of relying on the gradle-protobuf plugin.

Thoughts?

All information about this is available at https://docs.buf.build/bsr/remote-generation/remote-plugin-execution

**Comment by andrewparmet** [comment]
`bufGenerate` is not unavailable due to any requirement around protoc - it's unavailable because it didn't exist when I first wrote this plugin.

`bufGenerate` is on the list of features I'd like to add one day, but unfortunately I don't have the bandwidth to work on that right now. In the meantime you can define a custom Gradle task to do the same work, or I'm happy to work with you if you want to contribute the code yourself.

See https://github.com/andrewparmet/buf-gradle-plugin/issues/26#issuecomment-929818594 for my plans moving forward. I ran into https://github.com/andrewparmet/buf-gradle-plugin/issues/34, so the first step is to back out the changes to use symlinks and return to copying files into a "virtual" workspace for the original protobuf-gradle-plugin workflow.

**Comment by andrewparmet** [comment]
`buf generate` is now available in [0.4.1](https://github.com/andrewparmet/buf-gradle-plugin/releases/tag/0.4.1). Give it a try!
